.github/workflows/check-docs-links.yml

```diff
@@ -0,0 +1,19 @@
+name: Check links in docs
+on:
+  schedule:
+    # Run every day at 5:05am UTC
+    - cron:  '05 05 * * *'
```

Comment:
Decided to have it run on a schedule instead of on every PR.

- I would _hope_ folks are checking that new links they're adding are valid as part of the PR process.
- Making a bunch of HTTP requests to various sites is an error-prone process, and I don't want spurious HTTP failures to block the PR process, especially since most of the time the failure won't have anything to do with the PR in question.